scripts/cybershake/cybershake_progress.py

```diff
@@ -0,0 +1,320 @@
+    )
+
+    overall_df = progress_df.sum()
+
+    total_r_count = progress_df["total", R_COUNT_COL].sum()
+
+    f.write("\nOverall progress\n")
+
+    for proc_type in PROCESS_TYPES:
+        f.write(
+            _summary_template_fmt.format(
+                proc_type,
+                total_complete_rels[proc_type],
+                total_r_count,
+                total_complete_rels[proc_type] / total_r_count * 100,
+                overall_df[proc_type, ACT_CORE_HOURS_COL],
+                overall_df[proc_type, EST_CORE_HOURS_COL],
+                overall_df[proc_type, ACT_CORE_HOURS_COL]
+                / overall_df[proc_type, EST_CORE_HOURS_COL]
+                * 100,
+                overall_df[proc_type, EST_CORE_HOURS_COL]
+                - est_completed_df[proc_type].sum(),
+            )
+        )
+    f.close()
+
+
+def send2slack(msg_file, users, url):
+    with open(msg_file, "r") as f:
+        lines = f.readlines()
+
+    summary = (
+        " ".join(users) + " " + "\n".join(lines[-4:])
+    )  # .encode('utf-8') #last 4 lines
+    print(summary)
+    data = json.dumps({"text": summary})
+    urlopen(url, data=data.encode("utf-8"))
+
+
+def main(root_dir: Path, cybershake_list, outfile, slack_config=None):
+
+    runs_dir = Path(sim_struct.get_runs_dir(root_dir))
+    fault_names, r_counts = get_faults_and_r_count(cybershake_list)
+
+    sort_ind = np.argsort(fault_names)
+    fault_names, r_counts = fault_names[sort_ind], r_counts[sort_ind]
+
+    faults = np.asarray(
+        [sim_struct.get_fault_dir(root_dir, fault_name) for fault_name in fault_names]
+    )
+
+    if not root_dir.is_dir() or not runs_dir.is_dir():
+        print("Not a valid cybershake root directory. Quitting!")
+        return
+
+    # Create new progress df
+    progress_df = get_new_progress_df(root_dir, runs_dir, faults, fault_names, r_counts)
+
+    mgmtdb = MgmtDB(sim_struct.get_mgmt_db(root_dir))
+
+    write_progress(progress_df, outfile, mgmtdb)
+
+    if slack_config is not None:
+        send2slack(outfile, slack_config["users"], slack_config["url"])
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "cybershake_root", type=Path, help="The cybershake root directory"
+    )
+    parser.add_argument("list", type=str, help="The list of faults to run")
+    parser.add_argument("outfile", type=Path, help="output file")
+
+    parser.add_argument("--alert", help="send the output to slack", action="store_true")
+
+    args = parser.parse_args()
+
+    slack_config = None
+    slack_config_path = Path(__file__).resolve().parent / SLACK_ALERT_CONFIG
+    if args.alert:
+        try:
+            with open(slack_config_path) as f:
+                slack_config = load_yaml(f)
+        except FileNotFoundError:
+            print(
+                f"Error: --alert option used but {slack_config_path} not found. Exiting"
+            )
+            sys.exit(1)
```

Comment:
I think raising our own FileNotFoundError would be preferable to print and exit

Reply:
Replaced with assert